chainer/initializers/constant.py

```diff
@@ -17,10 +17,24 @@ class Identity(initializer.Initializer):
 
     """
 
-    def __init__(self, scale=1.0):
+    def __init__(self, scale=1.0, **kwargs):
+        super(Identity, self).__init__(**kwargs)
         self.scale = scale
 
-    def __call__(self, array):
+    def __call__(self, array=None, shape=None, xp=None):
+        if array is None:
+            assert isinstance(shape, tuple)
```

Comment:
How about add type check for `xp` as well? The same things may hold for other initializers below.